.github/workflows/aod-iam.yml

```diff
@@ -0,0 +1,51 @@
+# Workflow that validates and handles AOD IAM request.
+name: 'aod-iam'
+
+on:
+  pull_request:
+    types: ['opened']
+    branches:
+      - 'main'
+  pull_request_review:
+    types: ['submitted']
+
+jobs:
+  # Check if it is an AOD request
+  is_iam:
+    uses: 'abcxyz/aod-tempate/.github/workflows/reusable-checker.yml@main' # Since it does not have any access to github context, directly ping the tag/ref here.
+    with:
+      path: 'iam.yaml'
+
+  # Add always fail job.
+  always-fail:
+    needs: 'is_iam'
+    if: ${{ needs.is_iam.outputs.is_aod == 'true'}}
+    runs-on: ubuntu-latest
+    steps:
+      - id: 'fail'
+        run: exit 1
+
+  set_inputs:
+    needs: 'is_iam'
+    if: ${{ needs.is_iam.outputs.is_aod == 'true'}}
+    uses: 'abcxyz/aod-tempate/.github/workflows/reusable-set-inputs.yml@main'
```

Comment:
Not sure why this is needed? Can we use repo variable instead?

Reply:
We can, but then how users set those variables, either manually by themselves, or use a workflow. I tried to use workflow to set repo variables, it did not work since the GH token scope does not support variable related permission (will need to use PAT or token minter?).  Comparing to that, it seems way easier to do it with this set_inputs job.

Reply:
Users will have to set WIF stuff manually. I don't see a way around it

Reply:
I still think editing the set_inputs.yml file is easier comparing to mannually adding variables. Besides the former will have a PR for people to approve.

Reply:
This leads me to think maybe we can improve this and the set label workflow by using a config file. A centralized place to set the variables aod needs including durations. We can use another reusable workflow to parse the config and output variables and create labels.

Reply:
Let's make the basic functionality work before optimizing.

Reply:
discussed online.